explorer/components/Era5WrfXrayControls.vue

```diff
@@ -0,0 +1,73 @@
+<script lang="ts" setup>
+const startDate = defineModel<string>('startDate', { default: '' })
+const endDate = defineModel<string>('endDate', { default: '' })
+
+// Date validation
+const dateRangeError = computed(() => {
+  if (!startDate.value || !endDate.value) return null
+  if (startDate.value > endDate.value) {
+    return 'Start date must be before end date'
+  }
+  if (startDate.value < '1960-01-01') {
+    return 'Start date cannot be before 1960-01-01'
+  }
+  if (endDate.value > '2023-12-31') {
+    return 'End date cannot be after 2023-12-31'
+  }
+  return null
+})
+</script>
+
+<template>
+  <div class="controls panel">
+    <div class="panel-heading has-background-white">Select Date Range</div>
+
+    <div class="panel-block is-block">
```

Comment:
I think we need to pull this out.  The UX is easily broken (click on calendar, try and scroll year).  It would need a lot of nuance to function well and it has a bunch of pitfalls.

Let's discuss, but what if we offered a couple of predefined tasters, with a really loud thing saying "this is just the tip of the iceberg!  get it all by clicking here and do your own thing".  Then we can optimize the UX.